File: backend/src/main/java/ch/zuehlke/fullstack/hackathon/client/dto/InsightAddress.java
package ch.zuehlke.fullstack.hackathon.client.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class InsightAddress {
    @JsonProperty("Street")
    private String street;
    @JsonProperty("Zip")
    private String zip;
    @JsonProperty("City")
    private String city;
    @JsonProperty("Country")
    private String country;
}

File: frontend/src/app/shared/models/insight-address.ts
export class InsightAddress {
  private readonly _street: string;
  private readonly _zip: string;
  private readonly _city: string;
  private readonly _country: string;

  constructor(street: string, zip: string, city: string, country: string) {
    this._street = street;
    this._zip = zip;
    this._city = city;
    this._country = country;
  }

  get street(): string {
    return this._street;
  }

  get zip(): string {
    return this._zip;
  }

  get city(): string {
    return this._city;
  }

  get country(): string {
    return this._country;
  }
}
